dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java

```diff
@@ -0,0 +1,325 @@
+
+
+        File tempFile = File.createTempFile("testFileAsset2-es", ".jpg");
+        URL url = FocalPointAPITest.class.getResource("/images/test.jpg");
+        File testImage = new File(url.getFile());
+        FileUtils.copyFile(testImage, tempFile);
+        
+        testFileAsset2MultiLingual =APILocator.getFileAssetAPI().fromContentlet( new FileAssetDataGen(tempFile).languageId(testLanguage.getId()).folder(testFolder).nextPersisted());
+
+
+        
+        // archived
+        testFileAsset3Archived = APILocator.getFileAssetAPI().fromContentlet(FileAssetDataGen.createFileAsset(testFolder, "text3FileAsset", ".txt"));
+        APILocator.getContentletAPI().archive(testFileAsset3Archived, APILocator.systemUser(), false);
+        
+
+        testDotAsset =  TestDataUtils.getDotAssetLikeContentlet(testFolder);
+        
+        testSubFolder =  new FolderDataGen().name("testSubFolder").parent(testFolder).nextPersisted();
+
+        Template template=new Template();
+        template.setTitle("a template "+UUIDGenerator.generateUuid());
+        template.setBody("<html><body> I'm mostly empty </body></html>");
+        template=APILocator.getTemplateAPI().saveTemplate(template, testHost, APILocator.systemUser(), false);
+        
+        HTMLPageAsset page = new HTMLPageDataGen(testFolder, template).next();
+        page.setTitle("testingpage1");
+        testPage = APILocator.getHTMLPageAssetAPI().fromContentlet(HTMLPageDataGen.checkin(page, IndexPolicy.FORCE));
+
+        testlink = new LinkDataGen().hostId(testHost.getIdentifier()).title("testLink").parent(testFolder).target("https://google.com").linkType("EXTERNAL").nextPersisted();
+
+    }
+
+    
+
+
+
+    
+    @Test
+    public void testGetFolderContentWithInvalidIdentifier() { // https://github.com/dotCMS/core/issues/11829
+
+        final String NOT_EXISTING_ID = "01234567-1234-1234-1234-123456789012";
+
+        try {
+            browserAPI.getFolderContent( APILocator.systemUser(), NOT_EXISTING_ID, 0, -1, "", null, null, true, false, false, false, "", false, false, 1 );
+        } catch ( Exception e ){
+            Assert.assertTrue( e instanceof NotFoundInDbException );
+        }
+    }
+
+    @Test
+    public void testGetFolderContentWithValidIdentifier() throws Exception { // https://github.com/dotCMS/core/issues/11829
+
+        final String folderPath = "/BrowserAPITest-Folder";
+
+        //Creating folder to check.
+        User user = userAPI.getSystemUser();
+        Host demo = testHost;
+        Folder folder = folderAPI.createFolders( folderPath, demo, user, false );
+
+        try {
+            Map<String, Object> folderContent = browserAPI.getFolderContent( APILocator.systemUser(), folder.getInode(), 0, -1, "", null, null, true, false, false, false, "", false, false, 1 );
+            Assert.assertTrue( folderContent.containsKey( "total" ) );
+            Assert.assertTrue( folderContent.containsKey( "list" ) );
+        } catch ( Exception e ){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)